api/src/main/java/org/pmiops/workbench/cohortreview/ReviewTabQueryBuilder.java

```diff
@@ -13,92 +12,160 @@
+    String tableName = DomainType.MASTER.equals(domain)
+      ? MASTER_TABLE : TABLE_PREFIX + domain.toString().toLowerCase();
+    String finalSql = String.format(BASE_SQL_TEMPLATE + getSqlTemplate(domain) + FROM + WHERE_TEMPLATE,
+      tableName,
+      pageRequest.getSortColumn(),
+      pageRequest.getSortOrder().toString(),
+      pageRequest.getPageSize(),
+      pageRequest.getPage() * pageRequest.getPageSize());
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    params.put(NAMED_PARTICIPANTID_PARAM, QueryParameterValue.int64(participantId));
+    return QueryJobConfiguration
+      .newBuilder(finalSql)
+      .setNamedParameters(params)
+      .setUseLegacySql(false)
+      .build();
+  }
+
+  public QueryJobConfiguration buildCountQuery(Long participantId,
+                                               DomainType domain) {
+    String tableName = DomainType.MASTER.equals(domain)
+      ? MASTER_TABLE : TABLE_PREFIX + domain.toString().toLowerCase();
+    String finalSql = String.format(COUNT_TEMPLATE, tableName);
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    params.put(NAMED_PARTICIPANTID_PARAM, QueryParameterValue.int64(participantId));
+    return QueryJobConfiguration
+      .newBuilder(finalSql)
+      .setNamedParameters(params)
+      .setUseLegacySql(false)
+      .build();
+  }
+
+  public QueryJobConfiguration buildDetailsQuery(Long dataId, DomainType domain) {
+    return QueryJobConfiguration
+      .newBuilder("select * from `${projectId}.${dataSetId}.p_all_events` where person_id = -999")
+      .setUseLegacySql(false)
+      .build();
+  }
 
-    public QueryJobConfiguration buildDetailsQuery(Long dataId, DomainType domain) {
-        String domainString = domain.toString();
-        String tableName = DomainType.MASTER.equals(domain)
-          ? MASTER_TABLE : TABLE_PREFIX + domainString.toLowerCase();
-        String extraColumns = EXTRA_COLUMNS.get(domainString) == null
-          ? "" : EXTRA_COLUMNS.get(domainString);
-        String finalSql = String.format(SQL_TEMPLATE + DETAILS_WHERE, extraColumns, tableName);
-        Map<String, QueryParameterValue> params = new HashMap<>();
-        params.put(NAMED_DATAID_PARAM, QueryParameterValue.int64(dataId));
-        return QueryJobConfiguration
-          .newBuilder(finalSql)
-          .setNamedParameters(params)
-          .setUseLegacySql(false)
-          .build();
+  private String getSqlTemplate(DomainType domainType) {
+    switch (domainType) {
+      case MASTER:
+        return ALL_EVENTS_SQL_TEMPLATE;
+      case DRUG:
+        return DRUG_SQL_TEMPLATE;
+      case CONDITION:
+        return CONDITION_SQL_TEMPLATE;
+      case PROCEDURE:
+        return PROCEDURE_SQL_TEMPLATE;
+      case OBSERVATION:
+        return OBSERVATION_SQL_TEMPLATE;
+      case MEASUREMENT:
+        return MEASUREMENT_SQL_TEMPLATE;
+      default:
+        return PHYSICAL_MEASURE_SQL_TEMPLATE;
```

Comment:
Should the default be physical measurements?

Reply:
We can rethink this.